spatialdata/_core/_rasterize.py

```diff
@@ -0,0 +1,512 @@
+)
+from spatialdata._core.models import (
+    Image2DModel,
+    Image3DModel,
+    Labels2DModel,
+    Labels3DModel,
+    get_schema,
+)
+from spatialdata._core.transformations import (
+    Affine,
+    BaseTransformation,
+    Scale,
+    Sequence,
+    Translation,
+    _get_affine_for_element,
+)
+from spatialdata._types import ArrayLike
+from spatialdata.utils import Number, _parse_list_into_array
+
+
+def _compute_target_dimensions(
+    spatial_axes: tuple[str, ...],
+    min_coordinate: Union[list[Number], ArrayLike],
+    max_coordinate: Union[list[Number], ArrayLike],
+    target_unit_to_pixels: Optional[float],
+    target_width: Optional[float],
+    target_height: Optional[float],
+    target_depth: Optional[float],
+) -> tuple[float, float, Optional[float]]:
+    """
+    Compute the pixel sizes (width, height, depth) of the image that will be produced by the rasterization.
+
+    Parameters
+    ----------
+    spatial_axes
+        The axes that min_coordinate and max_coordinate refer to.
+    min_coordinate
+        The minimum coordinates of the bounding box.
+    max_coordinate
+        The maximum coordinates of the bounding box.
+    target_unit_to_pixels
+        The number of pixels per unit that the target image should have. It is mandatory to specify precisely one of
+        the following options: target_unit_to_pixels, target_width, target_height, target_depth.
+    target_width
+        The width of the target image in units. It is mandatory to specify precisely one of the following options:
+        target_unit_to_pixels, target_width, target_height, target_depth.
+    target_height
+        The height of the target image in units. It is mandatory to specify precisely one of the following options:
+        target_unit_to_pixels, target_width, target_height, target_depth.
+    target_depth
+        The depth of the target image in units. It is mandatory to specify precisely one of the following options:
+        target_unit_to_pixels, target_width, target_height, target_depth.
+
+    Returns
+    -------
+    target_width, target_height, target_depth
+        The pixel sizes (width, height, depth) of the image that will be produced by the rasterization.
+        If spatial_axes does not contain "z", target_depth will be None.
+    """
+    if isinstance(target_width, int):
+        target_width = float(target_width)
+    if isinstance(target_height, int):
+        target_height = float(target_height)
+    if isinstance(target_depth, int):
+        target_depth = float(target_depth)
+    assert (
+        np.sum(
+            [
+                target_unit_to_pixels is not None,
+                target_width is not None,
+                target_height is not None,
+                target_depth is not None,
+            ]
+        )
+        == 1
+    )
+    assert set(spatial_axes) == {"x", "y"} or set(spatial_axes) == {"x", "y", "z"}
+    if "z" not in spatial_axes:
+        assert target_depth is None
```

Comment:
I think these asserts should return an explanation (e.g., `assert x> 0, "x must be greater than 0`) so the user knows how to fix their issue.

```suggestion
    assert (
        np.sum(
            [
                target_unit_to_pixels is not None,
                target_width is not None,
                target_height is not None,
                target_depth is not None,
            ]
        )
        == 1
    ), "you must specify only one of: target_unit_to_pixels, target_width, target_height, target_depth" 
    assert set(spatial_axes) == {"x", "y"} or set(spatial_axes) == {"x", "y", "z"}
    if "z" not in spatial_axes:
        assert target_depth is None, "you cannot specify a target depth for 2D data"
```